x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetDataFrameAnalyticsStatsAction.java

```diff
@@ -193,26 +192,13 @@ public Stats(StreamInput in) throws IOException {
                 id = in.readString();
                 state = DataFrameAnalyticsState.fromStream(in);
                 failureReason = in.readOptionalString();
-                if (in.getVersion().before(Version.V_7_4_0)) {
-                    progress = readProgressFromLegacy(state, in);
```

Comment:
For this file it is probably worth leaving the BWC code in master for a bit longer.  It's very complex and would be a nasty shock for someone seeing it for the first time on a backport to 7.x.